tests/cpp/test_matmul_scheduler.cpp

```diff
@@ -2248,6 +2248,126 @@ TEST_F(MatmulSchedulerTest, StridedBatchEpilogueSingleBias) {
+  // sizes and re-use FusionExecutorCache)
+  for (auto layout : kAllSupportedMmaLayout) {
+    for (bool add_2d_bias : {false, true}) {
+      for (bool downcast_output : {false, true}) {
+        // TODO: maybe use bias
+        auto fusion = std::make_unique<Fusion>();
+        FusionGuard fg(fusion.get());
+
+        auto tv0 = makeContigTensor(2, DataType::Half);
+        auto tv1 = makeContigTensor(2, DataType::Half);
+        fusion->addInput(tv0);
+        fusion->addInput(tv1);
+
+        tv0 = canonicalizeInputToBMNK(tv0, layout, MmaOperand::A);
+        tv1 = canonicalizeInputToBMNK(tv1, layout, MmaOperand::B);
+        auto tv2 = fusedMultiplySum(tv0, tv1, {-1});
+
+        if (add_2d_bias) {
+          auto bias = makeContigTensor(2, DataType::Half);
+          fusion->addInput(bias);
+          tv2 = add(tv2, bias);
+        }
+
+        if (downcast_output) {
+          tv2 = castOp(DataType::Half, tv2);
+        }
+
+        fusion->addOutput(tv2);
+
+        const auto fusion_layout = mma_utils::getMmaLayout(fusion.get());
+        NVF_CHECK(
+            fusion_layout.isValid(),
+            "failed to get decide matmul layout through fusion definition");
+        NVF_CHECK(
+            fusion_layout.getData() == layout,
+            "mismatch between test layout (",
+            toString(layout),
+            ") and layout inferred from fusion definition (",
+            toString(fusion_layout.getData()),
+            ")");
+
+        FusionExecutorCache executor_cache(std::move(fusion));
+
+        for (const auto& [M, N, K, alignA, alignB, alignBias] :
+             std::vector<std::tuple<int, int, int, int, int, int>>{
+                 {504, 136, 248, 8, 8, 8}, // all fully vectorizable in all
+                                           // layouts
+                 {504, 136, 249, 8, 8, 8}, // odd K, operands not vectorizable
+                                           // in TN. output fully vectorizable
+                 {504, 137, 248, 8, 8, 8}, // A fully vectorizable, B fully
+                                           // vectorizable unless transposed,
+                                           // output not vectorizable
+                 {505, 136, 248, 8, 8, 8}, // B fully vectorizable, A
+                                           // vectorizable unless transposed,
+                                           // output fully vectorizable
+                 {505, 137, 248, 8, 8, 8}, // none vectorizable
+
+                 // The following cases are failing due to
+                 // https://github.com/NVIDIA/Fuser/issues/2118
+                 // Cases with vectorizable strides but misaligned base pointers
+                 //{504, 136, 248, 2, 8, 8}, // A not vectorizable due to offset
+                 //{504, 136, 248, 8, 2, 8}, // B not vectorizable due to offset
+                 //{504, 136, 248, 8, 8, 2}, // epilogue not vectorizable due to
+                 // offset
```

Comment:
Given that none of our other schedulers seem to support misaligned inputs (in the sense of having a base pointer not aligned to 16 bytes), that we should leave this for another PR.

Reply:
Note however that I did try replacing this https://github.com/NVIDIA/Fuser/blob/43370c6564c74995332f2642497563172fb3f2c8/csrc/kernel_cache.cpp#L1613-L1614
with 
```c++
 KernelArgumentHolder args_metadata = args; 
```
and these cases succeed. So I think once #2118 is addressed we can re-enable these cases.

Reply:
Waiting for #2123 to enable these.